### DISCUSSION TITLE
I immediately run into the barrier of preserving my important data while iterating forward on my data model

### DISCUSSION THREAD

**Discussion by pbiggar** [darklang/dark#4233] [Feedback]
A user says:
>I'll admit my initial reactions were pretty skeptical. Today I would generally say that my eyes are opened and I'm pretty optimistic about what y'all are doing here. 
This is a really impressive platform that I hope to be able to use more in the future.
>
>So off the bat, I want to say that you are really nailing the experience of "always live" applications that are easy to hack on. I love the explicit tooling around creating handlers, workers, etc while removing a lot of the cruft and boilerplate to get started. The trace-based development thing is pretty revolutionary IMO and really differentiates Dark from other platforms.
>
>One big part of the vision in that blog post is data migrations. That's a pretty big gap right now.The ability to easily create datastores that are already integrated with no fiddling around is also a big differentiator. But I immediately run into the barrier of preserving my important data while iterating forward on my data model. There aren't enough tools for that in Dark today.
Reading back over the blog post, the vision around data migrations sounds amazing. I hope that materializes soon.
>
>One area where I still have skepticism is feature flags. The workflow is a bit confusing today and doesn't feel great.
In the blog post, there's a strong case for leaning heavily on feature flags to enable seamless updates to code. I think I buy into this conceptually, but I don't think I can see how we get there from here.
>
>In this other thread, I started to draw a distinction between two modes of working.
Early on, you greenfielding and creating new things. And that code probably isn't getting traffic yet. This feels good in Dark today, and it's a good bet that it's what most of the early adopters are doing.[One] blog post targets a future where people have built complex apps in Dark that need to be maintained on an ongoing basis. That workflow and mindset is very different. Existing code needs to keep working at all times. And changes on go live to users when they've been tested and verified.
>
>It sounds like Dark's vision is that all of this work happens within the feature flags workflow, and that just doesn't feel good. Feature flags feel like a very cramped space to make minor updates rather than a first class citizen.
There are few tools for managing feature flags. It's difficult to do coarse branching of handlers and workers based on feature flags. It's not clear how data migrations should be handled along with feature flags. Which dovetails with the general lack of data migration support.

**Comment by pbiggar** [comment]
tl;dr really good qualitative discussion of feature flags and data migrations.
